superset-frontend/cypress-base/cypress/integration/dashboard/editmode.test.ts

```diff
@@ -119,6 +149,282 @@ describe('Dashboard edit', () => {
     cy.preserveLogin();
   });
 
+  describe('Color consistency', () => {
+    beforeEach(() => {
+      visitResetTabbedDashboard();
+    });
+
+    after(() => {
+      resetTabbedDashboard();
+    });
+
+    it('should respect chart color scheme when none is set for the dashboard', () => {
+      openProperties();
+      cy.get('[aria-label="Select color scheme"]').should('have.value', '');
+      applyChanges();
+      saveChanges();
+
+      // open nested tab
+      openTab(1, 1);
+      waitForChartLoad({
+        name: 'Top 10 California Names Timeseries',
+        viz: 'line',
+      });
+
+      // label Anthony
+      cy.get('.line .nv-legend-symbol')
+        .first()
+        .should('have.css', 'fill', 'rgb(31, 168, 201)');
+    });
+
+    it('should apply same color to same labels', () => {
+      openProperties();
+      selectColorScheme('lyftColors');
+      applyChanges();
+      saveChanges();
+
+      // open nested tab
+      openTab(1, 1);
+      waitForChartLoad({
+        name: 'Top 10 California Names Timeseries',
+        viz: 'line',
+      });
+
+      // label Anthony
+      cy.get('.line .nv-legend-symbol')
+        .first()
+        .should('have.css', 'fill', 'rgb(234, 11, 140)');
+
+      // open 2nd main tab
+      openTab(0, 1);
+      waitForChartLoad({ name: 'Trends', viz: 'line' });
+
+      // label Anthony
+      cy.get('.line .nv-legend-symbol')
+        .eq(2)
+        .should('have.css', 'fill', 'rgb(234, 11, 140)');
+    });
+
+    it('should change color scheme multiple times', () => {
+      openProperties();
+      selectColorScheme('lyftColors');
+      applyChanges();
+      saveChanges();
+
+      // open nested tab
+      openTab(1, 1);
+      waitForChartLoad({
+        name: 'Top 10 California Names Timeseries',
+        viz: 'line',
+      });
+
+      // label Anthony
+      cy.get('.line .nv-legend-symbol')
+        .first()
+        .should('have.css', 'fill', 'rgb(234, 11, 140)');
+
+      // open 2nd main tab
+      openTab(0, 1);
+      waitForChartLoad({ name: 'Trends', viz: 'line' });
+
+      // label Anthony
+      cy.get('.line .nv-legend-symbol')
+        .eq(2)
+        .should('have.css', 'fill', 'rgb(234, 11, 140)');
+
```

Comment:
Here need to call `editDashboard` before `openProperties`

Reply:
@stephenLYZ this is done on the `beforeEach` with `visitResetTabbedDashboard`

Reply:
Ye…But after saving the dashboard, it has already exited edit mode, because it is within the same test case and it will not call `beforeEach` again.

Reply:
You are right. I was looking at the wrong thing. Thanks!